parsons/action_network/action_network.py

```diff
@@ -0,0 +1,233 @@
+            "{'error': 'Couldn't find <object_name> with id = <id>'}"
+        """
+        return self._get_entry("people", object_id)
+
+    def add_person(self, email_address, given_name=None, family_name=None, tags=[],
+                   languages_spoken=[], postal_addresses=[],
+                   silent=False, **kwargs):
+        """
+        `Args:`
+            email_address:
+                Can be any of the following
+                    - a string with the person's email
+                    - a dictionary with the following fields
+                        - email_address (REQUIRED)
+                        - primary (OPTIONAL): Boolean indicating the user's primary email address
+                        - status (OPTIONAL): can taken on any of these values
+                            - "subscribed"
+                            - "unsubscribed"
+                            - "bouncing"
+                            - "previous bounce"
+                            - "spam complaint"
+                            - "previous spam complaint"
+            given_name:
+                The person's given name
+            family_name:
+                The person's family name
+            tags:
+                Any tags to be applied to the person
+            languages_spoken:
+                Optional field. A list of strings of the languages spoken by the person
+            postal_addresses:
+                Optional field. A list of dictionaries.
+                For details, see Action Network's documentation:
+                https://actionnetwork.org/docs/v2/people#put
+            silent:
+                If true, prints to the person's Action Network id when added successfully
+            **kwargs:
+                Any additional fields to store about the person. Action Network allows
+                any custom field.
+        Adds a person to Action Network
+        """
+        email_addresses_field = None
+        if type(email_address) == str:
+            email_addresses_field = [{"address": email_address}]
+        elif type(email_address == list):
+            if type(email_address[0]) == str:
+                email_addresses_field = [{"address": email} for email in email_address]
+                email_addresses_field[0]['primary'] = True
+            if type(email_address[0]) == dict:
+                email_addresses_field = email_address
+        if not email_addresses_field:
+            raise("email_address must be a string, list of strings, or list of dictionaries")
+        data = {
+            "person": {
+                "email_addresses": email_addresses_field,
+                "given_name": given_name,
+                "family_name": family_name,
+                "languages_spoken": languages_spoken,
+                "postal_addresses": postal_addresses,
+                "custom_fields": {**kwargs}
+              },
+            "add_tags": tags
+        }
+        self._add_entry("people", data, silent)
```

Comment:
Minor thing, but my preference is to get rid of the `_add_entry`, `_update_entry`, and `_get_entry` methods. I think the code is a little more readable if you can see inline what `_add_entry` is doing, and if you get rid of the `silent` argument for `_add_entry` and `_update_entry` they end up being a single line each, which is easy enough to just run here within this method.

`_get_entry_list` is doing a lot more, so it makes sense to have it be a separate method.